plasmapy/formulary/tests/test_thermal_speed.py

```diff
@@ -0,0 +1,401 @@
+)
+def test_aliases(alias, parent):
+    assert alias is parent
+
+
+class TestThermalSpeedCoefficients:
+    """
+    Test class for
+    `plasmapy.formulary.parameters.thermal_speed_coefficients`.
+    """
+
+    @pytest.mark.parametrize(
+        "ndim, method, _error",
+        [
+            (0, "most_probable", ValueError),
+            (4, "most_probable", ValueError),
+            ("not an int", "most_probable", ValueError),
+            (1, "not a valid method", ValueError),
+            (1, 2, ValueError),
+        ],
+    )
+    def test_raises(self, ndim, method, _error):
+        """Test scenarios that raise Exceptions."""
+        with pytest.raises(_error):
+            thermal_speed_coefficients(ndim=ndim, method=method)
+
+    @pytest.mark.parametrize(
+        "ndim, method, expected",
+        [
+            (1, "most_probable", 0),
+            (2, "most_probable", 1),
+            (3, "most_probable", np.sqrt(2)),
+            (1, "rms", 1),
+            (2, "rms", np.sqrt(2)),
+            (3, "rms", np.sqrt(3)),
+            (1, "mean_magnitude", np.sqrt(2 / np.pi)),
+            (2, "mean_magnitude", np.sqrt(np.pi / 2)),
+            (3, "mean_magnitude", np.sqrt(8 / np.pi)),
+            (1, "nrl", 1),
+            (2, "nrl", 1),
+            (3, "nrl", 1),
+        ],
+    )
+    def test_values(self, ndim, method, expected):
+        """Test that the correct values are returned."""
+        val = thermal_speed_coefficients(ndim=ndim, method=method)
+        assert np.isclose(val, expected)
+
+
+class TestThermalSpeed:
+    """
+    Test class for functionality of
+    `plasmapy.formulary.parameters.thermal_speed`, which include...
+
+    - Scenarios for raised exceptions
+    - Scenarios for issued warnings
+    - Basic behavior of `thermal_speed`
+    - Proper binding of Lite-Function functionality
+
+    Note: Testing of `thermal_speed_coefficients` and
+    `thermal_speed_lite` are done in separate test classes.
+    """
+
+    @pytest.mark.parametrize(
+        "bound_name, bound_attr",
+        [
+            ("lite", thermal_speed_lite),
+            ("coefficients", thermal_speed_coefficients),
+        ],
+    )
+    def test_lite_function_binding(self, bound_name, bound_attr):
+        """Test expected attributes are bound correctly."""
+        assert hasattr(thermal_speed, bound_name)
+        assert getattr(thermal_speed, bound_name) is bound_attr
+
+    def test_lite_function_marking(self):
+        """
+        Test thermal_speed is marked as having a Lite-Function.
+        """
+        assert hasattr(thermal_speed, "__bound_lite_func__")
+        assert isinstance(thermal_speed.__bound_lite_func__, dict)
+
+        for bound_name, bound_origin in thermal_speed.__bound_lite_func__.items():
+            assert hasattr(thermal_speed, bound_name)
+
+            attr = getattr(thermal_speed, bound_name)
+            origin = f"{attr.__module__}.{attr.__name__}"
+            assert origin == bound_origin
```

Comment:
Not sure how to parametrize this since this is checking that `__bound_lite_func__` was defined properly.  This is more like equating a dictionary than a parameterized test, and that's why it's written like this.  ... If you have a suggestion for a different approach then I'm open to it.  I just don't see how `pytest.mark.parametrize` could be used here, as much as I'd like to.